app/controllers/providers/proceedings_types_controller.rb

```diff
@@ -4,23 +4,26 @@ class ProceedingsTypesController < ProviderBaseController
 
     # GET /provider/applications/:legal_aid_application_id/proceedings_types
     def index
+      @multiple_proceedings = Setting.allow_multiple_proceedings?
       proceeding_types
     end
 
     # POST /provider/applications/:legal_aid_application_id/proceedings_types
     def create
       return continue_or_draft if draft_selected?
 
-      if legal_aid_application.proceeding_types.present?
+      if run_transaction
         go_forward
       else
         legal_aid_application.errors.add(:'proceeding-search-input', t('.search_and_select'))
+        @multiple_proceedings = Setting.allow_multiple_proceedings?
         proceeding_types
         render :index
       end
     end
 
     # PATCH /provider/applications/:legal_aid_application_id/proceedings_types/:id
+    # TODO: Could be removed after multiple proceedings go live.
```

Comment:
I'm thinking it might be better to call run_transaction here and use the `Setting.allow_multiple_proceedings?` to dictate how to process the proceeding type so this isn't missed. Like:

```
    ActiveRecord::Base.transaction do
      @legal_aid_application.reset_proceeding_types! unless Setting.allow_multiple_proceedings?
      @legal_aid_application.proceeding_types << proceeding_type(proceeding_type_id)
      AddScopeLimitationService.call(@legal_aid_application, scope_limitation)
    end
    ```
in your service file. Then you could also remove the `def proceeding_type` private method below